.circleci/config.yml

```diff
@@ -0,0 +1,182 @@
+
+      - run:
+          name: bundle install
+          command: |
+            bundle install
+
+      - save_cache:
+          name: save gem cache
+          paths:
+            - ./vendor/bundle
+          key: v1-bundle-<< parameters.ruby_version >>-{{ .Branch }}-{{ checksum "tiny_tds.gemspec" }}
+
+      - run:
+          name: build gem
+          command: |
+            bundle exec rake build
+
+      - run:
+          name: test gem
+          command: |
+            bundle exec rake test
+
+  test_windows:
+    parameters:
+      ruby_version:
+        description: "version tag for the veracross/circleci-ruby-freetds container"
+        type: string
+
+    executor:
+      name: win/server-2022
+      shell: powershell.exe
+
+    environment:
+      RAKEOPT: "-rdevkit"
+      TESTOPTS: '-v'
+      MAKE: "make V=1 -j2"
+
+    steps:
+      - run:
+          name: remove pre-installed ruby
+          command: |
+            Get-ChildItem -path 'C:\tools\' -filter Ruby* | Remove-Item -Force -Recurse
+
+      - run:
+          name: install ruby devkit
+          command: |
+            [Net.ServicePointManager]::SecurityProtocol = [Net.SecurityProtocolType]::Tls12;
+            Invoke-WebRequest -UseBasicParsing -uri "https://github.com/oneclick/rubyinstaller2/releases/download/RubyInstaller-<< parameters.ruby_version >>/rubyinstaller-devkit-<< parameters.ruby_version >>-x64.exe" -OutFile ruby-setup.exe;
+            .\ruby-setup.exe /VERYSILENT /NORESTART /DIR=C:/Ruby<< parameters.ruby_version >>-x64;
+
+      - run:
+          name: ruby diagnostics
+          command: |
+            $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
+            perl --version
+            ruby --version
+            gem --version
+            rm .\ruby-setup.exe
+
+      - run:
+          name: update build env (for ruby 2.5)
+          command: |
+            $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
+            ridk install 2 ;
+            gem install bundler
+
+      - checkout
+
+      - restore_cache:
+          name: restore gem cache
+          keys:
+            - v1-bundle-<< parameters.ruby_version >>-{{ .Branch }}-{{ checksum "tiny_tds.gemspec" }}
+            - v1-bundle-<< parameters.ruby_version >>-{{ .Branch }}-
+            - v1-bundle-<< parameters.ruby_version >>-
+
+      - run:
+          name: install dependencies
+          command: |
+            $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
+            bundle install
+
+      - save_cache:
+          name: save gem cache
+          paths:
+            - ./vendor/bundle
+          key: v1-bundle-<< parameters.ruby_version >>-{{ .Branch }}-{{ checksum "tiny_tds.gemspec" }}
+
+      - run:
+          name: build
+          no_output_timeout: 30m
+          command: |
+            $Env:PATH = "C:\\Ruby<< parameters.ruby_version >>-x64\\bin;$Env:PATH"
+            bundle exec rake ports
```

Comment:
Is it worth splitting this out so that each dependency is a separate step?

Reply:
can you clarify this?

Reply:
ohhh I see, yes I think I can do this